opnode/l2/ssz_test.go

```diff
@@ -0,0 +1,80 @@
+package l2
+
+import (
+	"bytes"
+	"encoding/binary"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/google/go-cmp/cmp"
+)
+
+// FuzzExecutionPayloadUnmarshal checks that our SSZ decoding never panics
+func FuzzExecutionPayloadUnmarshal(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		var payload ExecutionPayload
+		err := payload.UnmarshalSSZ(uint32(len(data)), bytes.NewReader(data))
+		if err != nil {
+			// not every input is a valid ExecutionPayload, that's ok. Should just not panic.
+			return
+		}
+	})
+}
+
+// FuzzExecutionPayloadMarshalUnmarshal checks that our SSZ encoding>decoding round trips properly
```

Comment:
What would it take to do differential fuzzing? i.e. compare our marshall / unmarshall against a known library

Reply:
Already implemented several libraries, including 2 Go libraries (and 3 different ways of representing the type structure), so there are ways... That said, fuzzing itself is good enough for initial code imho.